### ISSUE TITLE
azure_rm_virtualmachine_facts crashes on deallocated VMs

### ISSUE THREAD

**Issue by mooninite** [ansible/ansible#52181]
##### SUMMARY

Sometime overnight the storage disks metadata changed and the azure_rm_virtualmachine_facts module no longer works.

Setting the missing metadata to a static string allows the module to work again.

I have not debugged the returned metadata yet to see if this information is in a new location.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure_rm_virtualmachine_facts

##### ANSIBLE VERSION

```paste below
ansible 2.7.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/etc/ansible/custom_modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Sep 12 2018, 05:31:16) [GCC 4.8.5 20150623 (Red Hat 4.8.5-36)]
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

RHEL 7.6

##### DEPENDENCY PROBLEMS

```paste below

```

##### STEPS TO REPRODUCE

```yaml
- connection: local
  hosts: localhost
  gather_facts: false
  tasks:
  - name: facts test
    azure_rm_virtualmachine_facts:
      resource_group: YOURGROUPHERE
```

##### EXPECTED RESULTS

Facts returned

##### ACTUAL RESULTS

Stock module:
```
fatal: [localhost]: FAILED! => changed=false 
  module_stderr: |-
    No handlers could be found for logger "msrestazure.azure_active_directory"
    Traceback (most recent call last):
      File "<stdin>", line 113, in <module>
      File "<stdin>", line 105, in _ansiballz_main
      File "<stdin>", line 48, in invoke_module
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/__main__.py", line 340, in <module>
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/__main__.py", line 336, in main
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/__main__.py", line 224, in __init__
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/ansible_azure_rm_virtualmachine_facts_payload.zip/ansible/module_utils/azure_rm_common.py", line 308, in __init__
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/__main__.py", line 236, in exec_module
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/__main__.py", line 265, in list_items
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_DQY2Y3/__main__.py", line 321, in serialize_vm
    KeyError: 'diskSizeGB'
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1
```
After setting diskSizeGB to '0':
```
fatal: [localhost]: FAILED! => changed=false 
  module_stderr: |-
    No handlers could be found for logger "msrestazure.azure_active_directory"
    Traceback (most recent call last):
      File "<stdin>", line 113, in <module>
      File "<stdin>", line 105, in _ansiballz_main
      File "<stdin>", line 48, in invoke_module
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/__main__.py", line 340, in <module>
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/__main__.py", line 336, in main
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/__main__.py", line 224, in __init__
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/ansible_azure_rm_virtualmachine_facts_payload.zip/ansible/module_utils/azure_rm_common.py", line 308, in __init__
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/__main__.py", line 236, in exec_module
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/__main__.py", line 265, in list_items
      File "/tmp/ansible_azure_rm_virtualmachine_facts_payload_XKJ8me/__main__.py", line 322, in serialize_vm
    KeyError: 'storageAccountType'
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1
```

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @gustavomcarmo @haroldwongms @nitzmahone @trstringer @xscript @yuwzho @zikalino
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@mooninite, just so you are aware we have a dedicated Working Group for azure.
You can find other people interested in this in `#ansible-azure` on Freenode IRC
For more information about communities, meetings and agendas see https://github.com/ansible/community

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by Fred-sun** [comment]
@mooninite Thanks for your reporting this issue! Do you add new data disk(Type: StandardSSD_LRS) for the VM when you get the VM facts?

**Comment by mooninite** [comment]
The VMs have not been changed. I saw this bug in the middle of a scheduled job. It had worked for the first few stages and failed in the final stages so Microsoft introduced a bug on us.

Unfortunately I can no longer reproduce this. Microsoft appears to have fixed their bug and the metadata returned is now valid and the module works.

**Comment by mooninite** [comment]
I saw this issue again over the weekend and I understand why I saw it now. A few days ago I stopped "stopping" our VMs and started "deallocating" VMs to save money. The VMs being deallocated are returning less metadata for dataDisks and this is causing the azure_rm_virtualmachine_facts module to crash the way it is. Should the module perform a second lookup to find the data disks that have been deallocated or should the module return "deallocated" in place of disk data?

Example:
* VM powered on or stopped:
"dataDisks": [{"diskSizeGB": 128, "createOption": "attach", "name": "foo-datavg01-pv01", "caching": "None", "managedDisk": {"storageAccountType": "Premium_LRS", "id": "/subscriptions/uuid/resourceGroups/resource-group/providers/Microsoft.Compute/disks/foo-datavg01-pv01"}, "lun": 0}]}
* VM deallocated:
"dataDisks": [{"caching": "None", "managedDisk": {"id": "/subscriptions/uuid/resourceGroups/resource-group/providers/Microsoft.Compute/disks/foo-datavg01-pv01"}
